packages/langium/src/lsp/hover-provider.ts

```diff
@@ -48,13 +51,36 @@ export abstract class AstNodeHoverProvider implements HoverProvider {
                 if (targetNode) {
                     return this.getAstNodeHoverContent(targetNode);
                 }
+
+                // Add support for documentation on keywords
+                if (isKeyword(cstNode.grammarSource)) {
+                    return this.getKeywordHoverContent(cstNode.grammarSource);
+                }
             }
         }
         return undefined;
     }
 
     protected abstract getAstNodeHoverContent(node: AstNode): MaybePromise<Hover | undefined>;
 
+    protected getKeywordHoverContent(node: AstNode): MaybePromise<Hover | undefined> {
+        let comment = isAstNodeWithComment(node) ? node.$comment : undefined;
+        if (!comment) {
+            comment = findCommentNode(node.$cstNode, ['ML_COMMENT'])?.text;
+        }
+        if (comment && isJSDoc(comment)) {
+            const content = parseJSDoc(comment).toMarkdown();
```

Comment:
@praisethemoon For specific use-cases it is always possible to override this method and provide a different logic

```ts
const content = parseJSDoc(comment).getTag('hover')?.content.toMarkdown();
```

Reply:
I like this! Thanks for the tip :)